cpp/src/dual_simplex/branch_and_bound.cpp

```
@@ -480,6 +601,31 @@ mip_status_t branch_and_bound_t<i_t, f_t>::set_final_solution(mip_solution_t<i_t
+#if 1
+    FILE* fid = NULL;
+    fid       = fopen("solution.dat", "w");
+    if (fid != NULL) {
+      printf("Writing solution.dat\n");
+
+      std::vector<f_t> residual = original_lp_.rhs;
+      matrix_vector_multiply(original_lp_.A, 1.0, incumbent_.x, -1.0, residual);
+      printf("|| A*x - b ||_inf %e\n", vector_norm_inf<i_t, f_t>(residual));
+      auto hash_combine_f = [](size_t seed, f_t x) {
+        seed ^= std::hash<f_t>{}(x) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+        return seed;
```

Cppcheck (2.19.0)

[error] Resource leak

(resourceLeak)